### DISCUSSION TITLE
ca65/ld65: Overwrite + Fill = ??

### DISCUSSION THREAD

**Discussion by TheRealQuantam** [cc65/cc65#2054] [Q&A]
How exactly do the segment type = overwrite and the memory fill = yes options interact in ld65? I notice whenever I make a memory region (representing a bank) filled when there are overwrite segments in it I get really weird binary file sizes (too large). Then when I make it not filled I have to manually pad it to be the right size for the bank, looking at the map file to see how much padding is needed...

**Comment by bbbradsmith** [comment]
This is likely due to the bugs identified #1366 which should be addressed as of #1995

Ultimately it's probably not worth trying to describe how those interact prior to that change, but I think at this point it should be more intuitive. Overwrite segments can be applied with or without fill, and if fill=no the overwriting segment is still allowed to extend the length of the output memory region if needed, as long as it doesn't exceed its total capacity.
